deployment/configenv/xml_jlibpt/SWProcess.cpp

```diff
@@ -41,22 +41,25 @@ SWProcess::SWProcess(const char* name, EnvHelper * envHelper):SWComponentBase(na
 
 IPropertyTree * SWProcess::addComponent(IPropertyTree *params)
 {
+  IPropertyTree * pCompTree;
   const char* clone = params->queryProp("@clone");
   if (clone)
   {
-     return SWComponentBase::cloneComponent(params);
+     pCompTree = SWComponentBase::cloneComponent(params);
```

Comment:
The clone here seems to be redundant with the check for a clone in the new code above that calls cloneComponent (lines 33-37 in SWEspProcess.cpp. In keeping with OO, it would seem that the addComponent and cloneComponent methods should do exactly what their names imply and that a decision to call cloneComponent should be made prior to calling.

Reply:
Since envgen2 only expose "add" but not "clone" action later is included in the first.
"add" action has three scenarios:
1) the component already exists. In this case "add" will add/update attributes as well as possible add a sub-element based user input in "add" action
2) the component doesn't exist but user want to clone an existing component by keeping the setting in that component
3) the component doesn't exist and user want to create a new component based xsd default values and attributes provided in "add" action.

For most software component "addComponent" function is enough. But for EspProcess extra bindings need applied which is why it is overriding here.